### PR TITLE
ZCS-5707 Updating seniority index for HAB group

### PR DESCRIPTION
**Pull request by rupalid**
Added operation to update seniority index to Modify HABGroup
 <urn1:ModifyHABGroupRequest>
         <urn1:habGroupOperation habGroupId="id"  seniorityIndex="34" op="assignSeniority"/>
 </urn1:ModifyHABGroupRequest>

Added zmprov command for updating seniority index of HAB group:
mhabsi  {habGrpId} {seniorityIndex}

Bulk updates can be done using
zmprov < update.txt
where update.txt contains:
mhabsi {id1} {seniorityIndex}
mhabsi {id2} {seniorityIndex}
mhabsi {id2} {seniorityIndex}

### CODE REVIEW

File: store/src/java/com/zimbra/cs/account/ProvUtil.java [hunk 1]
```diff
@@ -852,17 +852,19 @@ public void printHelp() {
...
-            "{domain} {ouName} {newName}", Category.MISC , 3, 3),
+            "{domain} {ouName} {newName}", Category.HAB , 3, 3),
         DELETE_HAB_OU("deleteHABOrgUnit", "dhou",
-            "{domain} {ouName}", Category.MISC , 2, 2),
+            "{domain} {ouName}", Category.HAB , 2, 2),
         CREATE_HAB_GROUP("createHabGroup", "chabg",
-            "{groupName} {ouName} {name@domain} {TRUE|FALSE} [attr1 value1 [attr2 value2...]]", Category.MISC , 3, Integer.MAX_VALUE),
+            "{groupName} {ouName} {name@domain} {TRUE|FALSE} [attr1 value1 [attr2 value2...]]", Category.HAB , 3, Integer.MAX_VALUE),
         GET_HAB("getHab", "ghab",
             "{habRootGrpId} ", Category.ACCOUNT , 1, 1),
```
Comment (desouzas) [inline store/src/java/com/zimbra/cs/account/ProvUtil.java]: `Category.HAB`?

File: store/src/java/com/zimbra/cs/account/soap/SoapProvisioning.java [hunk 2]
```diff
@@ -3175,4 +3175,18 @@ public void modifyHabGroup(String habGroupId, String currentParentGroupId,
         invokeJaxb(new ModifyHABGroupRequest(operation));
     }
 
+    /**
+     * 
+     * @param habGroupId HAB group to be modified
+     * @param seniorityIndex seniority index
+     * @throws ServiceException if an aeeror occurs while modifying the HAB group
```
Comment (desouzas) [inline store/src/java/com/zimbra/cs/account/soap/SoapProvisioning.java]: `error`

File: soap/src/java/com/zimbra/soap/admin/type/HABGroupOperation.java [hunk 3]
```diff
@@ -69,9 +70,16 @@ public static HabGroupOp fromString(String s) throws ServiceException {
...
+                                                                      */, required = false)
     private String targetHabGroupId;
-    
+
+    /**
+     * @zm-api-field-tag seniorityIndex
+     * @zm-api-field-description the seniorityInex of HAB group
+     */
+    @XmlAttribute(name = AccountConstants.A_HAB_SENIORITY_INDEX /* op */, required = false)
+    private int seniorityIndex;
```
Comment (desouzas) [inline soap/src/java/com/zimbra/soap/admin/type/HABGroupOperation.java]: Not required, default to 0 is expected?

File: store/src/java/com/zimbra/cs/account/ProvUtil.java (hunk 1 above)
Comment (rupalid) [reply store/src/java/com/zimbra/cs/account/ProvUtil.java]: Fixed

File: store/src/java/com/zimbra/cs/account/soap/SoapProvisioning.java (hunk 2 above)
Comment (rupalid) [reply store/src/java/com/zimbra/cs/account/soap/SoapProvisioning.java]: fixed

File: soap/src/java/com/zimbra/soap/admin/type/HABGroupOperation.java (hunk 3 above)
Comment (rupalid) [reply soap/src/java/com/zimbra/soap/admin/type/HABGroupOperation.java]: Fixed

**Pull request merged**